chalice/config.py

```diff
@@ -79,13 +80,15 @@ class Config(object):
 
     def __init__(self,
                  chalice_stage=DEFAULT_STAGE_NAME,
+                 function_name=DEFAULT_HANDLER_NAME,
                  user_provided_params=None,
                  config_from_disk=None,
                  default_params=None):
-        # type: (str, StrMap, StrMap, StrMap) -> None
+        # type: (str, str, StrMap, StrMap, StrMap) -> None
         #: Params that a user provided explicitly,
         #: typically via the command line.
         self.chalice_stage = chalice_stage
+        self.function_name = function_name
```

Comment:
So I think this is fine for now as lambda functions are really the only thing we configure. But how do you see this will expand if we ever support per-resource configurations for other AWS resources besides Lambda? Would we add resource name to the config as well and the config would then be scoped to: stage + lambda function + any other specific managed resource configuration? I just want to make sure there won't be any issues expanding this out